portfolio/src/main/webapp/comments.js

```diff
@@ -117,19 +117,14 @@ function deleteComment(comment, messageDiv) {
   const params = new URLSearchParams();
   params.append('id', comment.id);
   fetch('/delete-comment', {method: 'POST', body: params})
-  .then(response => response.json())
-  .then(userInfo => 
+  .then(response => 
   {
-    if (userInfo.loggedIn) {
-      if (userInfo.userId === (comment.userId)) {
-        
+    var status = response.status;
+    if (status == 200) {
         //delete only possible with login and same id as author
         messageDiv.remove(); 
-      } else {
-        alert("You can only delete your own comment(s).");
-      }
     } else {
-      window.location = userInfo.loginUrl;
+      alert("You can only delete your own comment(s), and you must login.");
```

Comment:
Unless you are specifically parsing the error message to determine that this is the problem, don't assume that it's the problem.  Otherwise, when you're wrong, you'll mislead users.

Reply:
Added multiple additional HTTP status code responses to specify.